jarbas/api/tests/test_reimbursement_view.py

```diff
@@ -123,76 +89,52 @@ def test_status(self):
 
     def test_contents(self):
         contents = loads(self.resp.content.decode('utf-8'))
-        expected = dict(
-            applicant_id=13,
-            batch_number=9,
-            cnpj_cpf='11111111111111',
-            congressperson_document=2,
-            congressperson_id=1,
-            congressperson_name='Roger That',
-            document_id=42,
-            document_number='6',
-            document_type=7,
-            document_value=8.90,
-            installment=7,
-            issue_date='1970-01-01',
-            leg_of_the_trip='8',
-            month=1,
-            party='Partido',
-            passenger='John Doe',
-            all_reimbursement_numbers=[10, 11],
-            all_reimbursement_values=[12.13, 14.15],
-            all_net_values=[1.99, 2.99],
-            remark_value=1.23,
-            state='UF',
-            subquota_description='Subquota description',
-            subquota_group_description='Subquota group desc',
-            subquota_group_id=5,
-            subquota_id=4,
-            supplier='Acme',
-            term=1970,
-            term_id=3,
-            total_net_value=4.56,
-            total_reimbursement_value=None,
-            year=1970,
-            probability=0.5,
-            suspicions=suspicions,
-            receipt=dict(fetched=False, url=None)
-        )
-        self.assertEqual(expected, contents)
+        for result_attr, result_value in contents.items():
+            if not hasattr(self.reimbursement, result_attr):
+                continue
+            expected_value = getattr(self.reimbursement, result_attr)
+            self.assertEqual(str(result_value), str(expected_value))
```

Comment:
@cuducos, good points. I'll gona try do not be too extense.

I do not agree completly about that. The test here should test that each API response attribute is equal to the object attribute in database. It's what being done properly. Of course:

```
response = load(api_response)
mydict = dict(one=1, two=2)
assert mydict == mydict
```

is in a first glance such more obvious than:

```
response = load(api_response)
mydict = dict(one=1 two=2)
for key, value in mydict.items():
	return assert response[key] == value
```

However I still think the positive consequences are higher than negative ones. Think about changing (for any reason) the response content structure (not values). If we change an attribute name or even any attribute data type. We'll have to adapt the tests only because we change some implementation specificity, not a business rule, nor an application requirement. On the other hand, if we change some business logic that makes API response different from object attribute value, the test still will fail.

I think the tests should be abstract and resilient enought to test requirements properly without being a pain for developers.

It's just my thoughts, let me know what you think.

ps.: about this topic, I would recomend this post: http://blog.cleancoder.com/uncle-bob/2017/05/05/TestDefinitions.html